### PR TITLE
feat: nestable names

### PR DESCRIPTION
**Pull request by FauxFaux**
Some support for namespaces.

I hate that Ctx object but can't see how to fix it.

### CODE REVIEW

File: src/ctx/locales.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    pub fn assign_local(&mut self, n: impl ToString, v: Word) -> Result<()> {
+        let n = n.to_string();
+        if n.contains('_') {
+            return Err(JError::NonceError).context("namespaced names");
+        }
+
+        self.anon
+            .last_mut()
+            .ok_or(JError::DomainError)
+            .context("local assignment with no local scope")?
```
Comment (AshyIsMe) [inline src/ctx/locales.rs:67]: This actually shouldn't be an error.

"Local" scope also means the current locale if the current point is not within a verb:

```j
$ ijconsole
   a=.42
   a
42
   a_base_
42

```

Comment (AshyIsMe) [reply src/ctx/locales.rs:67]: Locales are also how j does object oriented programming, I think I'm happy to just kick that can down the road unless you want to dig into it now?

- https://www.jsoftware.com/help/jforc/modular_code.htm#_Toc191734489
- https://code.jsoftware.com/wiki/Vocabulary/Locales#Working_with_classes_and_instances

Comment (FauxFaux) [reply src/ctx/locales.rs:67]: I fixed `=.` at the global scope by adding a new anonymous namespace at the top by default, no idea if this is a good way to model it.

Comment (AshyIsMe) [reply src/ctx/locales.rs:67]: It'll do for now.  By default the repl lands in the `base` namespace in j:
- https://code.jsoftware.com/wiki/Vocabulary/Locales#The_base_locale_and_the_z-locale

But let's merge this and fix that later.

**Pull request merged**